### PR TITLE
Add icon checklist workflow

### LINKED ISSUE #2070: Checklist for self-review
### About this discussion

I want to change #1318 a little bit so that it is useful, but at the same time easy to implement.

Due to the fact that difficulties with taking into account the Lawnicons guidelines occur after several PRs, I suggest making a checklist for self-review. This checklist should be sent via **github-actions** as the first comment on all PRs that have icons.

Such a checklist can be useful for both beginners and contributors who rarely add icons (to remind them of the main points of the guidelines). It will also save some time for reviewers, because they will be less likely to repeat themselves.

**Suggestion**

```
Hey! Thanks for the contribution. While there is no reviewer, you can do a self-review. This will help to fix the reasons why the icons may not be suitable for Lawnicons. 

### Canvas and sizes [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#canvas--sizes)
1. Canvas: `192×192px`.
2. Non-square icons: the long side of the icons should be `160px`.
3. Square icons: `154×154px`.

### Color, stroke width and rounding [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#foundation)
1. Color: black `#000`.
2. No fill. Base stroke width: `12px`. `14px`, `10px`, `8px` — depending on the shape of the icons. `6px` — for fine details.
3. Rounded ends and joins. 90° corners are rounded by `6-32px`.

### Naming [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#naming)
1. Names should match the official ones and contain no additional text.
2. If the original name is not in English, then add a localized (or transliterated) one to it using `~~`.
Example: `フールー ~~ Hulu`.
3. The names of the drawables should repeat the names of the apps if nothing prevents it.

### Quality [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#details)
1. Icons should be easily recognizable.
2. Align icons to [the visual center](https://www.google.com/search?sca_esv=1fcec3f5e0b15e20&q=what+is+visual+center+of+an+icon&tbm=isch&source=lnms).
3. No blackness. Reduce the stroke width or simplify the icons so that there are no noticeable black spots.
4. Avoid close distances between strokes. The icons on the phone screen will be smaller, so the small distances between the strokes will stick together.
5. Avoid drastic changes in stroke widths. When the strokes next to each other differ in width by 4px or more, the icon will look sloppy.

```

### PR DESCRIPTION
**Pull request by Hamster45105**
## Description

Basis for this change discussed in  #2070 

## How the action works

When a PR is opened involving icons, `github-actions` will leave a comment. This comment can be found and edited in `.github/icon_checklist.md`

Here is an example of it running on a PR: https://github.com/Hamster45105/lawnicons/pull/4

## Note

I removed the `first_pr_interaction.yml` file since it would not really be needed if this action was implemented.

Fixes #2070 

## Type of change

:x: Bug fix (non-breaking change which fixes an issue)
:white_check_mark: General change (non-breaking change that doesn't fit the above categories, such as copyediting)

### CODE REVIEW

File: .github/workflows/icon_checklist.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Comment icon checklist
+
+on:
+    pull_request:
+      branches: [develop]
+      types: [opened]
+      paths:
+        - svgs/*
+        - app/assets/appfilter.xml
```
Comment (x9136) [inline .github/workflows/icon_checklist.yml:9]: When a contributor adds only links, they don't need a comment about self-review.

**Comment by x9136** [comment]
> are you happy for me to mark as ready for review now?

It's up to you. In any case, someone else will do the code review.

File: .github/icon_checklist.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Hey! Thanks for the contribution. While there is no reviewer, you can do a self-review. This will help to fix the reasons why the icons may not be suitable for Lawnicons.
```
Comment (SuperDragonXD) [inline .github/icon_checklist.md:1]:
```suggestion
Thanks for your contribution!

While waiting for a review from our team, you can do a self-review to ensure that your icons are suitable for Lawnicons.
```

File: .github/icon_checklist.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Hey! Thanks for the contribution. While there is no reviewer, you can do a self-review. This will help to fix the reasons why the icons may not be suitable for Lawnicons. 
+
+### Canvas and sizes [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#canvas--sizes)
```
Comment (SuperDragonXD) [inline .github/icon_checklist.md:3]: Using `→` isn't really accessible. You could replace it with `(more info)` instead or simply make the whole heading a link.

This goes to all of the headers using `→`.

File: .github/icon_checklist.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+### Naming [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#naming)
+1. Names should match the official ones and contain no additional text.
+2. If the original name is not in English, then add a localized (or transliterated) one to it using `~~`.
+Example: `フールー ~~ Hulu`.
+3. The names of the drawables should repeat the names of the apps if nothing prevents it.
+
+### Quality [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#details)
+1. Icons should be easily recognizable.
+2. Align icons to [the visual center](https://www.google.com/search?sca_esv=1fcec3f5e0b15e20&q=what+is+visual+center+of+an+icon&tbm=isch&source=lnms).
+3. No blackness. Reduce the stroke width or simplify the icons so that there are no noticeable black spots.
```
Comment (SuperDragonXD) [inline .github/icon_checklist.md:22]:
```suggestion
3. Avoid blackness or noticable black spots by reducing the stroke width or simplifying the icons.
```

File: .github/icon_checklist.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+3. Rounded ends and joins. 90° corners are rounded by `6-32px`.
+
+### Naming [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#naming)
+1. Names should match the official ones and contain no additional text.
+2. If the original name is not in English, then add a localized (or transliterated) one to it using `~~`.
+Example: `フールー ~~ Hulu`.
+3. The names of the drawables should repeat the names of the apps if nothing prevents it.
+
+### Quality [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#details)
+1. Icons should be easily recognizable.
```
Comment (SuperDragonXD) [inline .github/icon_checklist.md:20]:
```suggestion
1. Ensure that icons are easily recognizable.
```

File: .github/icon_checklist.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
+Hey! Thanks for the contribution. While there is no reviewer, you can do a self-review. This will help to fix the reasons why the icons may not be suitable for Lawnicons. 
+
+### Canvas and sizes [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#canvas--sizes)
+1. Canvas: `192×192px`.
+2. Non-square icons: the long side of the icons should be `160px`.
```
Comment (SuperDragonXD) [inline .github/icon_checklist.md:5]:
```suggestion
2. Non-square icons: the length of the icons should be `160px`.
```

File: .github/icon_checklist.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+2. If the original name is not in English, then add a localized (or transliterated) one to it using `~~`.
+Example: `フールー ~~ Hulu`.
+3. The names of the drawables should repeat the names of the apps if nothing prevents it.
+
+### Quality [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#details)
+1. Icons should be easily recognizable.
+2. Align icons to [the visual center](https://www.google.com/search?sca_esv=1fcec3f5e0b15e20&q=what+is+visual+center+of+an+icon&tbm=isch&source=lnms).
+3. No blackness. Reduce the stroke width or simplify the icons so that there are no noticeable black spots.
+4. Avoid close distances between strokes. The icons on the phone screen will be smaller, so the small distances between the strokes will stick together.
+5. Avoid drastic changes in stroke widths. When the strokes next to each other differ in width by 4px or more, the icon will look sloppy.
```
Comment (SuperDragonXD) [inline .github/icon_checklist.md:24]:
```suggestion
5. Avoid drastic changes in stroke widths. When the strokes next to each other differ in width by 4px or more, the icon will look sloppy.

```
I'd recommend adding a newline at the end of the file.

File: .github/workflows/icon_checklist.yml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const ICON_CHECKLIST = fs.readFileSync('.github/icon_checklist.md', 'utf8');
+            const issue = { owner: context.repo.owner, repo: context.repo.repo, issue_number: context.issue.number };
+            await github.rest.issues.createComment({
+              ...issue,
+              body: ICON_CHECKLIST
+            });
```
Comment (SuperDragonXD) [inline .github/workflows/icon_checklist.yml:27]:
```suggestion
            });

```
Same goes for here. Please add a newline.

File: .github/icon_checklist.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+2. Non-square icons: the long side of the icons should be `160px`.
+3. Square icons: `154×154px`.
+
+### Color, stroke width and rounding [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#foundation)
+1. Color: black `#000`.
+2. No fill. Base stroke width: `12px`. `14px`, `10px`, `8px` — depending on the shape of the icons. `6px` — for fine details.
+3. Rounded ends and joins. 90° corners are rounded by `6-32px`.
+
+### Naming [→](https://github.com/LawnchairLauncher/lawnicons/blob/develop/CONTRIBUTING.md#naming)
+1. Names should match the official ones and contain no additional text.
```
Comment (SuperDragonXD) [inline .github/icon_checklist.md:14]:
```suggestion
1. Names should match the official app name and contain no additional text.
```

**Review by SuperDragonXD — COMMENTED** [review]
Thanks for the PR and apologies for the delay.

The code is fine, but I have several suggestions regarding the icon checklist.

File: .github/icon_checklist.md (hunk 3 above)
Comment (x9136) [reply .github/icon_checklist.md:3]: They are specially made like this. It is necessary that they do not stand out too much relative to more relevant information for self-review. In terms of accessibility, it is enough that they are large and stand out in color.

File: .github/icon_checklist.md (hunk 6 above)
Comment (x9136) [reply .github/icon_checklist.md:5]: > the length of the icons should be `160px`.

The requirement is that the width or height should be 160px. In my understanding, "length" refers only to width.

File: .github/icon_checklist.md (hunk 3 above)
Comment (SuperDragonXD) [reply .github/icon_checklist.md:5]: I'll try to see any other alternative solution soon-ish, but sure, this will work for now.

**Review by SuperDragonXD — APPROVED** [review]

File: .github/icon_checklist.md (hunk 3 above)
Comment (x9136) [reply .github/icon_checklist.md:5]: I had the idea not to link to individual sections, but to link only to the guide. We can use it. For example, at the end, write "The Lawnicons guidelines".

Comment (SuperDragonXD) [reply .github/icon_checklist.md]: Sure thing. Here's an example:

```markdown
### More information
For a complete guide on the icon guidelines, visit the [Lawnicons Contributing Guidelines](CONTRIBUTING.md).
```

Comment (x9136) [reply .github/icon_checklist.md]: I deleted it in the end, because the link to the full guidelines is in the PR description template.

**Comment by x9136** [comment]
Great job, thanks!

**Pull request merged**